packages/hierarchies-react/src/presentation-hierarchies-react/internal/TreeActions.ts

```diff
@@ -284,12 +284,16 @@ export class TreeActions {
   }
 }
 
-function collectTreePartsUntil(untilNotifier: Observable<void>, rootNode?: TreeModelRootNode) {
+function collectTreePartsUntil(untilNotifier: Observable<void>, parent: TreeModelHierarchyNode | TreeModelRootNode, rootNode?: TreeModelRootNode) {
   return (source: Observable<LoadedTreePart>) =>
     source.pipe(
       reduce<LoadedTreePart, TreeModel>(
         (treeModel, loadedPart) => {
-          addNodesToModel(treeModel, loadedPart);
+          if ("loadedNodes" in loadedPart) {
+            addNodesToModel(treeModel, loadedPart.loadedNodes, loadedPart.parentId);
+          } else {
+            addErrorToModel(treeModel, loadedPart.error, parent);
+          }
```

Comment:
I don't like that we take `parent` as arg, and get `parentId` an attribute of `loadedPart`. `addNodesToModel` takes id from `loadedPart`, but `addErrorToModel` takes `parent` arg.

The two should match, but who knows... what could we do to get rid of one of them?

Reply:
Also, maybe we should consolidate `addNodesToModel` + `addErrorToModel` to `addTreePartToModel`?

Reply:
The problem is that the `treeModel` may not contain parent in its `idToNode`, in this case the error would be lost as we need to update that part, therefor we need to get parent from outside. For the `AddNodesToModel` it does not care what the parent is it just needs `parentId` so we can just pass LoadedPart spread out (we need it spread out because the loadedPart definition includes error). The only change here I think could make this cleaner is to have `addNodesToModel` take its own definition of `loadedPart` which does not include error.

Reply:
> The problem is that the `treeModel` may not contain parent in its `idToNode`, in this case the error would be lost as we need to update that part, therefor we need to get parent from outside.

Could `LoadedTreePart` contain `parent` instead of `parentId`?

Reply:
changed to load parent instead of parentId